### PR TITLE
fix: sync `PageMeta.title` with `PageBlockModel.title` while importing markdown files

### LINKED ISSUE #5296: Importing markdown files doesn't update the `name` in `PageMeta` and `WorkspaceMeta` correctly
I'm digging into https://github.com/toeverything/AFFiNE/issues/4906.

And found the cause is

1.  Here create an anonymous page for importing a markdown file.

https://github.com/toeverything/blocksuite/blob/ac6d17102bd630571add91be79ec70bf8a14ff8c/packages/blocks/src/_common/widgets/linked-page/import-page/import-page.ts#L38

2. Here forgot to update `name` in `PageMeta`

https://github.com/toeverything/blocksuite/blob/ac6d17102bd630571add91be79ec70bf8a14ff8c/packages/blocks/src/_legacy/content-parser/index.ts#L640-L642

https://github.com/toeverything/blocksuite/blob/ac6d17102bd630571add91be79ec70bf8a14ff8c/packages/blocks/src/_legacy/content-parser/index.ts#L991-L999

### PR DESCRIPTION
**Pull request by hyf0**
Fixes #5296. Thank @lawvs for guidance.

### CODE REVIEW

**Comment by fourdim** [comment]
Hi, could you please open a new pull request to add your name directly to the https://github.com/toeverything/blocksuite/edit/master/.github/CLA.md so that we can review this PR?

**Comment by hyf0** [comment]
> Hi, could you please open a new pull request to add your name directly to the https://github.com/toeverything/blocksuite/edit/master/.github/CLA.md so that we can review this PR?

Sure. https://github.com/toeverything/blocksuite/pull/5298

File: packages/blocks/src/_legacy/content-parser/index.ts [hunk 1]
```diff
@@ -996,6 +996,13 @@ export class ContentParser {
         options: tags,
       },
     });
+
+    // Make sure the title is synced with the model
+    const pageMetaTitle = this._page.meta.title;
+    const pageModelTitle = (this._page.root as PageBlockModel).title.toString();
+    if (pageMetaTitle === pageModelTitle) {
```
Comment (fourdim) [inline packages/blocks/src/_legacy/content-parser/index.ts:1003]: The logic here seems not correct. Maybe `!==`?

File: packages/blocks/src/_legacy/content-parser/index.ts [hunk 2]
```diff
@@ -996,6 +996,13 @@ export class ContentParser {
         options: tags,
       },
     });
+
+    // Make sure the title is synced with the model
+    const pageMetaTitle = this._page.meta.title;
+    const pageModelTitle = (this._page.root as PageBlockModel).title.toString();
+    if (pageMetaTitle !== pageModelTitle) {
+      this._page.workspace.meta.setName(pageModelTitle);
```
Comment (lawvs) [inline packages/blocks/src/_legacy/content-parser/index.ts:1004]: This name is not the page title, try `workspace.setPageMeta(pageId, { title })`

**Review by lawvs — CHANGES_REQUESTED** [review]

File: packages/blocks/src/_legacy/content-parser/index.ts [hunk 3]
```diff
@@ -996,6 +996,15 @@ export class ContentParser {
         options: tags,
       },
     });
+
+    // Make sure the title is synced with the model
+    const pageMetaTitle = this._page.meta.title;
+    const pageModelTitle = (this._page.root as PageBlockModel).title.toString();
```
Comment (fourdim) [inline packages/blocks/src/_legacy/content-parser/index.ts:1002]: Could you please try something like `if (this._page.root instanceof PageBlockModel)` to guard this?

**Comment by fourdim** [comment]
This PR will be approved and merged once the CLA PR is approved.

**Review by lawvs — APPROVED** [review]
It works, thank you!

https://github.com/toeverything/blocksuite/assets/18554747/a6a0b515-29b7-4920-9a0b-8d6b7f0bf74d

**Pull request merged**